pkg/webhook/policy_benchmark_test.go

```diff
@@ -0,0 +1,332 @@
+	for i := 0; i < M; i++ {
+		r := crList[i%len(crList)]
+		result[i] = *(r.DeepCopy())
+		r.SetName(genRandString(10))
+	}
+	return result
+}
+
+func genRandString(n int) string {
+	out := make([]byte, n)
+	for i := 0; i < n; i++ {
+		c := 'a' + rand.Intn(26)
+		out[i] = byte(c)
+	}
+	return string(out)
+}
+
+func createAdmissionRequests(resList []unstructured.Unstructured, n int) atypes.Request {
+	dryRun := false
+	res := resList[n%len(resList)]
+	name := "res-name-" + strconv.Itoa(n)
+	namespace := "res-namespace-" + strconv.Itoa(n)
+	res.SetName(name)
+	res.SetNamespace(namespace)
+	oldRes := res.DeepCopy()
+	res.SetResourceVersion("2")
+	oldRes.SetResourceVersion("1")
+	gvr, _ := meta.UnsafeGuessKindToResource(oldRes.GroupVersionKind())
+	return atypes.Request{
+		AdmissionRequest: admissionv1beta1.AdmissionRequest{
+			UID:                types.UID(uuid.NewUUID()),
+			Kind:               metav1.GroupVersionKind{Group: oldRes.GroupVersionKind().Group, Version: oldRes.GroupVersionKind().Version, Kind: oldRes.GroupVersionKind().Kind},
+			Resource:           metav1.GroupVersionResource{Group: gvr.Group, Version: gvr.Version, Resource: gvr.Resource},
+			SubResource:        "",
+			RequestKind:        &metav1.GroupVersionKind{Group: oldRes.GroupVersionKind().Group, Version: oldRes.GroupVersionKind().Version, Kind: oldRes.GroupVersionKind().Kind},
+			RequestResource:    &metav1.GroupVersionResource{Group: gvr.Group, Version: gvr.Version, Resource: gvr.Resource},
+			RequestSubResource: "",
+			Name:               name,
+			Namespace:          namespace,
+			Operation:          "UPDATE",
+			UserInfo: authenticationv1.UserInfo{
+				Username: "res-creator",
+				UID:      "uid",
+				Groups:   []string{"res-creator-group"},
+				Extra:    map[string]authenticationv1.ExtraValue{"extraKey": {"value1", "value2"}}},
+			Object:    runtime.RawExtension{Object: &resList[n%len(resList)]},
+			OldObject: runtime.RawExtension{Object: oldRes},
+			DryRun:    &dryRun,
+			Options:   runtime.RawExtension{},
+		},
+	}
+}
+
+func BenchmarkValidationHandler(b *testing.B) {
+	// setup test
+	opa, err := makeOpaClient()
+	if err != nil {
+		b.Fatalf("could not initialize OPA: %s", err)
+	}
+
+	c := &fakeNsGetter{scheme: scheme.Scheme, NoopClient: testclient.NoopClient{}}
+	cfg := &v1alpha1.Config{
+		Spec: v1alpha1.ConfigSpec{
+			Validation: v1alpha1.Validation{
+				Traces: []v1alpha1.Trace{},
+			},
+		},
+	}
+	h := validationHandler{opa: opa, client: c, injectedConfig: cfg}
+
+	// M constraints among T templates
+	// M is divided into T, 2T, 10T, 20T...100000T
+	benchmarks := map[string]struct {
+		// description of the test
+		description string
+		// directory to load constraint templates from
+		templateDir string
+		// directory to load constraints from
+		constraintDir string
+		// directory to load resources to be evaluated // ex. Pods
+		resourceDir string
+		// number of constraints to load
+		load []int
+	}{
+		"psp": {
```

Comment:
nit: we should name this based off the description... maybe `psp: 100% violations` or similar

Reply:
done

Reply:
changed